lib/phil/charlie_cards/importer.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Phil.CharlieCards.Importer do
+  @moduledoc """
+  Imports CharlieCard inventory into Phil.
+  """
+
+  import Ecto.Query, only: [from: 2]
+
+  alias NimbleCSV.RFC4180, as: CSV
+
+  alias Phil.CharlieCards.CharlieCard
+  alias Phil.Products.Product
+  alias Phil.Repo
+
+  @media %{"2" => :ticket, "5" => :card}
+
+  @doc """
+  Given the path of a file, this function attempts to import CharlieCards.
+
+  It returns a map of successes and failures like %{error: [], ok: []}.
+  Error entries are changesets and success entries are the inserted struct.
+  """
+  def import(path) do
+    path
+    |> File.stream!()
+    |> CSV.parse_stream()
+    |> Stream.map(&row_to_changeset/1)
+    |> Stream.chunk_every(100)
+    |> Task.async_stream(&insert_batch/1, max_concurrency: 10, ordered: false)
```

Comment:
do we need it to be async?

Reply:
Is there a downside? This will chunk the number of rows into groups of 100 and then use 10 processes to process those groups until they're all done.

Reply:
Well I kind of think we should start without asynchronicity and only add it in if it solves a problem for us. 
If it's to gain performance I would use insert_all first (it's probably faster). It's not necessarily wrong per se, but to me async always adds complexity and failure cases that can be annoying to debug / reason about. Can add memory overhead too because we have to copy data to other processes and back.